src/server/cluster/cluster_family.cc

```diff
@@ -502,10 +502,18 @@ void ClusterFamily::DflyClusterConfig(CmdArgList args, ConnectionContext* cntx)
     before = tl_cluster_config->GetOwnedSlots();
   }
 
-  auto cb = [&](util::ProactorBase* pb) { tl_cluster_config = new_config; };
+  DispatchTracker tracker{server_family_->GetListeners(), cntx->conn()};
+  auto cb = [&tracker, &new_config](util::ProactorBase* pb) {
+    tl_cluster_config = new_config;
+    tracker.TrackOnThread();
```

Comment:
why do we need TrackOnThread?
why TrackAll after AwaitFiberOnAll is finished to update the config on all threads is not good?

Reply:
Why not do it correctly if it takes only a few lines 🙂 

--- 

> TrackAll after AwaitFiberOnAll is finished to update the config on all threads is not good

Because it's possibly waiting for totally different stuff to finish, including commands that would already be running with the new cluster config